src/main/java/core/basesyntax/service/impl/FileWriterImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.FileWriter;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+
+public class FileWriterImpl implements FileWriter {
+
+    private final File file;
+
+    public FileWriterImpl(String fileName) {
+        this.file = new File(fileName);
+    }
+
+    @Override
+    public void write(String report) {
+        try (BufferedWriter writer = new BufferedWriter(new java.io.FileWriter(file))) {
+            writer.write(report);
+        } catch (IOException e) {
+            throw new RuntimeException("can't save report", e);
```

Comment:
Consider including the file path in the exception message to provide more context, e.g., "Can't save report to file: " + file.getPath().